ludwig/modules/lr_scheduler.py

```diff
@@ -0,0 +1,161 @@
+import logging
+import math
+from typing import Any, Dict
+
+from torch.optim import Optimizer
+from torch.optim.lr_scheduler import LambdaLR, ReduceLROnPlateau
+
+from ludwig.constants import MINIMIZE, TRAINING, VALIDATION
+from ludwig.modules.metric_modules import get_metric_cls
+from ludwig.schema.lr_scheduler import LRSchedulerConfig
+from ludwig.utils.metric_utils import TrainerMetric
+from ludwig.utils.trainer_utils import ProgressTracker
+
+
+class ReduceLROnPlateauLimited(ReduceLROnPlateau):
+    def __init__(self, optimizer: Optimizer, mode: str, reduce_limit: int, factor: float, patience: int):
+        super().__init__(optimizer, mode=mode, factor=factor, patience=patience)
+        self.reduce_limit = reduce_limit
+        self._num_reduce_lr = 0
+
+    def step(self, metrics, epoch=None):
```

Comment:
It looks like `epoch` is deprecated. Let's exclude it: https://pytorch.org/docs/stable/_modules/torch/optim/lr_scheduler.html#ReduceLROnPlateau